internal/app.go

```diff
@@ -300,6 +300,7 @@ func (a *App) checkin(client *http.Client, crypto CryptoHandler) error {
 	if err != nil {
 		return err
 	}
+	req.Header.Add("User-Agent", "fioconfig-2")
```

Comment:
IIRC we've had this same thing in `fioctl`, and then we decided that returning the official version based on the git tag is a better way to handle this due to its flexibility and observability benefits.

Reply:
i'm on the fence with this as we are using this to help backward api compatibility and not general observability. The problem if we just take a git tag is that we get into the world of understanding what tags support what level of API breakage, where a increasing "API compatibility" header makes for a pretty easy check.

The LmP version is a monotonic increasing version. Problem is - the tag is created *after* the release, so the binary doesn't have the version.

Long way of saying - i have no clear thought on what to put in this field.

Reply:
IMHO (but just IMHO) the fioconfig versions needs not necessarily be tied up to the lmp version.

That said, when we want to upgrade the fioconfig version in lmp - we can cut off the new tag for fioconfig, and reference it [here in lmp-meta](https://github.com/foundriesio/meta-lmp/blob/main/meta-lmp-base/recipes-support/fioconfig/fioconfig_git.bb#L14) just like we currently reference the commit hash (or define a different field for the version).

(In the Makefile I would simply do it similarly to how we do it in fioctl, so we could test version increase locally).

I have no strong opinion on this; it just seems more professional if the `fioconfig --version` outputs _exactly the same version_ as what we put into the `User-Agent` header. Ideally, it should be possible to backtrack that thing into the exact git commit (or tag).

Reply:
That said, from the simplicity perspective what you did is just _enough good for now_; so lazy mode is totally acceptable.

Reply:
i'm not against a `fioconfig --version` command. but what we are trying to accomplish here is more akin to something like the docker registry api where you are telling the server what you can accept. binding that to tags is hard long term (although a change like this only happens once every 2 years or so).

maybe UserAgent is the wrong thing or we need something extra. UserAgent could be the exact version, but then we need an "ApiVersion" type header that the backend can use.

Reply:
i just force pushed a different way that i think is more in spirit with user-agents and works nicely in the backend. lets agree on the device-gateway change, and then we can merge or change this.

Reply:
The User-Agent is now something parseable, so I think it is good enough for the starter.